Classes/Base/Locale.php

```diff
@@ -34,29 +34,35 @@ public static function get(\Aimeos\MShop\ContextIface $context,
     {
         if (!isset(self::$locale)) {
             $config = $context->config();
-
-
             $sitecode = $config->get('mshop/locale/site', 'default');
-            $name = $config->get('typo3/param/name/site', 'site');
-
-            if ($request !== null && $request->hasArgument($name) === true) {
-                $sitecode = $request->getArgument($name);
-            }
-
-
             $lang = $config->get('mshop/locale/language', '');
-            $name = $config->get('typo3/param/name/language', 'locale');
-
-            if ($request !== null && $request->hasArgument($name) === true) {
-                $lang = $request->getArgument($name);
-            }
-
-
             $currency = $config->get('mshop/locale/currency', '');
-            $name = $config->get('typo3/param/name/currency', 'currency');
 
-            if ($request !== null && $request->hasArgument($name) === true) {
-                $currency = $request->getArgument($name);
+            if ($request !== null) {
+                $name = $config->get('typo3/param/name/site', 'site');
+                $siteAttribute = $request->getAttribute('site');
+
+                if ($request->hasArgument($name) === true) {
+                    $sitecode = $request->getArgument($name);
+                } elseif ($siteAttribute && $siteAttribute->getSettings()->has('aimeosSite')) {
+                    $sitecode = $siteAttribute->getSettings()->get('aimeosSite');
+                }
```

Comment:
This can be written in a shorter way and with a default value:
```
} else {
    $sitecode = $request->getAttribute('site')?->getSettings()->get('aimeos.site', 'default');
}
```

Reply:
``` php
$sitecode = $request->getAttribute('site')?->getSettings()->get('aimeos.site') ?? 'default';
```
i would change it to this, so if site attribute is not set the fallback is 'default'

Reply:
According to the SiteSettings API, `get()` accepts the default parameter as second argument:
https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/SiteHandling/AccessingSiteConfiguration.html#the-php-sitesettings-object

Reply:
```php
$site = $request->getAttribute('site')?->getSettings()->get('aimeos.site', 'default');
```
$site could be null, if getAttribute('site') returns null.
So i would remove the nullsafe operator `?->` and let it throw or keep it like this:
```php
$sitecode = $request->getAttribute('site')?->getSettings()->get('aimeos.site') ?? $sitecode;
```

https://github.com/aimeos/aimeos-typo3/blob/d74c8f6da0be23a8db259e6a624cdef7d72e69ee/Classes/Base/Locale.php#L37-L39

Im using those values as fallback, otherwise the configuration from typoscript has no effect.

Reply:
You are right, this is correct due to the null-safe operator:
```
$sitecode = $request->getAttribute('site')?->getSettings()->get('aimeos.site') ?? $sitecode;
```